ambry-account/src/main/java/com/github/ambry/account/CompositeAccountService.java

```diff
@@ -0,0 +1,191 @@
+    try {
+      Container secondaryResult = secondaryAccountService.getContainer(accountName, containerName);
+      if (++getContainerRequestCount % 100 < config.samplingPercentageForGetContainerConsistencyCheck
+          && primaryResult != null && !primaryResult.equals(secondaryResult)) {
+        logger.warn("Inconsistency detected between primary and secondary for accountName ={}, containerName = {}",
+            accountName, containerName);
+        accountServiceMetrics.getContainerDataInconsistencyCount.inc();
+      }
+    } catch (Exception e) {
+      accountServiceMetrics.getContainerDataInconsistencyCount.inc();
+      logger.error("get container failed for secondary for accountName={}, containerName={}", accountName,
+          containerName, e);
+    }
+    return primaryResult;
+  }
+
+  @Override
+  public Set<Container> getContainersByStatus(Container.ContainerStatus containerStatus) {
+    Set<Container> primaryResult = primaryAccountService.getContainersByStatus(containerStatus);
+    Set<Container> secondaryResult = secondaryAccountService.getContainersByStatus(containerStatus);
+    if (!primaryResult.equals(secondaryResult)) {
+      logger.warn(
+          "Inconsistency detected between primary and secondary for containers with status ={}, primary ={}, secondary = {}",
+          containerStatus, primaryResult, secondaryResult);
+      accountServiceMetrics.getContainerDataInconsistencyCount.inc();
+    }
+    return primaryResult;
+  }
+
+  @Override
+  public void selectInactiveContainersAndMarkInZK(StatsSnapshot statsSnapshot) {
+    primaryAccountService.selectInactiveContainersAndMarkInZK(statsSnapshot);
+    try {
+      secondaryAccountService.selectInactiveContainersAndMarkInZK(statsSnapshot);
+    } catch (Exception e) {
+      logger.warn("Marking containers as inactive failed for secondary", e);
+      accountServiceMetrics.updateContainerInconsistencyCount.inc();
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    try {
+      primaryAccountService.close();
+    } finally {
+      secondaryAccountService.close();
+    }
+    shutDownExecutorService(scheduler, config.consistencyCheckerShutdownTimeoutMinutes, TimeUnit.MINUTES);
+  }
+
+  /**
+   * Compares and logs differences (if any) in Account metadata stored in primary and secondary sources
+   */
+  private void compareAccountMetadata() {
+    Set<Account> primaryAccounts = new HashSet<>(primaryAccountService.getAllAccounts());
+    Set<Account> secondaryAccounts = new HashSet<>(secondaryAccountService.getAllAccounts());
+    if (!primaryAccounts.equals(secondaryAccounts)) {
+      accountServiceMetrics.accountMetadataInconsistencyCount.inc();
+      logger.debug("Inconsistency detected between primary and secondary at {} for accounts ={}",
+          SystemTime.getInstance().milliseconds(), primaryAccounts.removeAll(secondaryAccounts));
```

Comment:
Note that `primaryAccounts.removeAll()` returns a boolean value rather than inconsistent accounts set.